core/src/Kokkos_View.hpp

```diff
@@ -132,6 +132,66 @@ void runtime_check_rank_host(const size_t dyn_rank, const bool is_void_spec,
 }
 #endif
 
+template <typename Traits, typename Layout>
+KOKKOS_INLINE_FUNCTION void runtime_check_rank(
+    Layout& layout, const std::string& label) {
+  size_t i0 = layout.dimension[0];
+  size_t i1 = layout.dimension[1];
+  size_t i2 = layout.dimension[2];
+  size_t i3 = layout.dimension[3];
+  size_t i4 = layout.dimension[4];
+  size_t i5 = layout.dimension[5];
+  size_t i6 = layout.dimension[6];
+  size_t i7 = layout.dimension[7];
+
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+  KOKKOS_IMPL_IF_ON_HOST
+  Impl::runtime_check_rank_host(
+      Traits::rank_dynamic,
+      std::is_same<typename Traits::specialize, void>::value, i0, i1, i2, i3,
+      i4, i5, i6, i7, label);
+  else Impl::runtime_check_rank_device(
+      Traits::rank_dynamic,
+      std::is_same<typename Traits::specialize, void>::value, i0, i1, i2, i3,
+      i4, i5, i6, i7);
+#else
+  Impl::runtime_check_rank_host(
+      Traits::rank_dynamic,
+      std::is_same<typename Traits::specialize, void>::value, i0, i1, i2, i3,
+      i4, i5, i6, i7, label);
+#endif
+}
+
+template <typename Traits, typename Layout>
+KOKKOS_INLINE_FUNCTION void runtime_check_rank(
+    Layout& layout) {
```

Comment:
lets remove this one and just call the other one with an empty label if necessary

Reply:
Ok, simplified the call graph. Also removed templating.